modules/samples/firealarm/component/api/src/main/java/org/homeautomation/firealarm/api/DeviceTypeService.java

```diff
@@ -0,0 +1,119 @@
+* Version 2.0 (the "License"); you may not use this file except
+* in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package org.homeautomation.firealarm.api;
+
+import org.homeautomation.firealarm.api.dto.DeviceJSON;
+
+import org.wso2.carbon.apimgt.annotations.api.API;
+import org.wso2.carbon.apimgt.annotations.api.Permission;
+import org.wso2.carbon.device.mgt.extensions.feature.mgt.annotations.DeviceType;
+import org.wso2.carbon.device.mgt.extensions.feature.mgt.annotations.Feature;
+
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.Path;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Produces;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.PUT;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+
+/**
+ * This is the controller API which is used to control agent side functionality
+ */
+@SuppressWarnings("NonJaxWsWebServices")
+@API(name = "firealarm", version = "1.0.0", context = "/firealarm", tags = "firealarm")
+@DeviceType(value = "firealarm")
+public interface DeviceTypeService {
+
+    /**
+     * @param agentInfo device owner,id and sensor value
+     * @return
+     */
+    @Path("device/register")
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Permission(scope = "firealarm_user", permissions = {"/permission/admin/device-mgt/user/register"})
+    Response registerDevice(final DeviceJSON agentInfo);
+
+    /**
+     * @param deviceId unique identifier for given device type
+     * @param state    change status of sensor: on/off
+     * @param response
+     */
+    @Path("device/{deviceId}/change-status")
+    @POST
+    @Feature(code = "change-status", name = "Change status of buzzer: on/off",
+            description = "Change status of buzzer: on/off")
+    @Permission(scope = "firealarm_user", permissions = {"/permission/admin/device-mgt/change-status"})
+    Response changeStatus(@PathParam("deviceId") String deviceId,
+                          @QueryParam("state") String state,
+                          @Context HttpServletResponse response);
+
+    /**
+     * Retrieve Sensor data for the given time period
+     * @param deviceId unique identifier for given device type instance
+     * @param sensorName name of the sensor
+     * @param from  starting time
+     * @param to    ending time
+     * @return  response with List<SensorRecord> object which includes sensor data which is requested
```

Comment:
Complete the parts of these comments where applicable, i.e. method descriptions, input param descriptions, returned value descriptions and descriptions of excpetions

Also add fullstops to the end of each description.